components/director/internal/domain/runtime/selfreg_config.go

```diff
@@ -0,0 +1,64 @@
+package runtime
+
+import (
+	"time"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/oauth"
+	"github.com/tidwall/gjson"
+)
+
+// SelfRegConfig is configuration for the runtime self-registration flow
+type SelfRegConfig struct {
+	SelfRegisterDistinguishLabelKey string `envconfig:"APP_SELF_REGISTER_DISTINGUISH_LABEL_KEY"`
+	SelfRegisterLabelKey            string `envconfig:"APP_SELF_REGISTER_LABEL_KEY,optional"`
+	SelfRegisterLabelValuePrefix    string `envconfig:"APP_SELF_REGISTER_LABEL_VALUE_PREFIX,optional"`
+	SelfRegisterResponseKey         string `envconfig:"APP_SELF_REGISTER_RESPONSE_KEY,optional"`
+	SelfRegisterPath                string `envconfig:"APP_SELF_REGISTER_PATH,optional"`
+	SelfRegisterNameQueryParam      string `envconfig:"APP_SELF_REGISTER_NAME_QUERY_PARAM,optional"`
+	SelfRegisterTenantQueryParam    string `envconfig:"APP_SELF_REGISTER_TENANT_QUERY_PARAM,optional"`
+	SelfRegisterRequestBodyPattern  string `envconfig:"APP_SELF_REGISTER_REQUEST_BODY_PATTERN,optional"`
+
+	OAuthMode      oauth.AuthMode `envconfig:"APP_SELF_REGISTER_OAUTH_MODE,default=oauth-mtls"`
+	OauthTokenPath string         `envconfig:"APP_SELF_REGISTER_OAUTH_TOKEN_PATH,optional"`
+
+	SkipSSLValidation bool `envconfig:"APP_SELF_REGISTER_SKIP_SSL_VALIDATION,default=false"`
+
+	ClientTimeout time.Duration `envconfig:"default=30s"`
+
+	InstanceClientIDPath     string                    `envconfig:"APP_SELF_REGISTER_INSTANCE_CLIENT_ID_PATH"`
+	InstanceClientSecretPath string                    `envconfig:"APP_SELF_REGISTER_INSTANCE_CLIENT_SECRET_PATH"`
+	InstanceURLPath          string                    `envconfig:"APP_SELF_REGISTER_INSTANCE_URL_PATH"`
+	InstanceTokenURLPath     string                    `envconfig:"APP_SELF_REGISTER_INSTANCE_TOKEN_URL_PATH"`
+	InstanceCertPath         string                    `envconfig:"APP_SELF_REGISTER_INSTANCE_X509_CERT_PATH"`
+	InstanceKeyPath          string                    `envconfig:"APP_SELF_REGISTER_INSTANCE_X509_KEY_PATH"`
+	InstanceConfigs          string                    `envconfig:"APP_SELF_REGISTER_AGGREGATED_INSTANCE_CONFIGS"`
+	RegionToInstanceConfig   map[string]InstanceConfig `envconfig:"-"`
+}
+
+// InstanceConfig is configuration for communication with specific instance in the runtime self-registration flow
+type InstanceConfig struct {
+	ClientID     string
+	ClientSecret string
+	URL          string
+	TokenURL     string
+	Cert         string
+	Key          string
+}
+
+// MapInstanceConfigs parses the InstanceConfigs json string to map with key: regin_name and value: InstanceConfig for the instance in the region
+func (c *SelfRegConfig) MapInstanceConfigs() {
+	bindingsResult := gjson.Parse(c.InstanceConfigs)
+	bindingsMap := bindingsResult.Map()
+	c.RegionToInstanceConfig = make(map[string]InstanceConfig)
+	for region, config := range bindingsMap {
+		i := InstanceConfig{
```

Comment:
Additional to Dimitar's comment about usage of gjson.Valid, can we ensure somehow(with validate function or something else) that the end result of `c.RegionToInstanceConfig` is what we expected? E.g. `bindigsMap` won't be empty + `gjson.Get` for the properties return non empty value? 

Example: `gjson.Get(config.String(), c.InstanceClientSecretPath).String()` tries from config's json to get client_secret from some path and that returns Result and e.g we can check that client_secret is actually exist with Result's `.Exists` method. 

That may help us to mitigate issues related to wrong path and missing property which from other hand most probably will result in error when getting token